ios/gamma/split-view/RNSSplitViewScreenController.swift

```diff
@@ -69,46 +87,144 @@ public class RNSSplitViewScreenController: UIViewController {
 
   // MARK: Layout
 
+  ///
+  /// @brief This method is overridden to extract the value to which we're transitioning
+  /// and attach the DisplayLink to track frame updates on the presentation layer.
+  ///
+  public override func viewWillTransition(
+    to size: CGSize,
+    with coordinator: any UIViewControllerTransitionCoordinator
+  ) {
+    super.viewWillTransition(to: size, with: coordinator)
+
+    transitionInProgress = true
+
+    coordinator.animate(
+      alongsideTransition: { [weak self] context in
+        guard let self = self else { return }
+        if self.displayLink == nil {
+          self.displayLink = CADisplayLink(
+            target: self, selector: #selector(trackTransitionProgress))
+          self.displayLink?.add(to: .main, forMode: .common)
+        }
+      },
+      completion: { [weak self] context in
+        guard let self = self else { return }
+        self.stopAnimation()
+        // After the animation completion, ensure that ShadowTree state
+        // is calculated relatively to the ancestor's frame by requesting
+        // the state update.
+        self.updateShadowTreeState()
+      })
+  }
+
+  private func stopAnimation() {
```

Comment:
This method is misnamed. It does not stop any animation. It only cleanups some temporary state & invalidates the display link. The name should reflect that.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3176